### PR TITLE
pass the correct dropout prob when using FlashAttention

### PR DESCRIPTION
**Pull request by ktaebum**
A recent fix at cdd2afdf22a8d3490b5c9091c957449765519fdc performs stateful modification of the dropout probability of the Attention layer.
This could lead to incorrect behavior after the first evaluation performed during training.

### CODE REVIEW

File: megatron/model/transformer.py [hunk 1]
```diff
@@ -383,13 +383,15 @@ def forward(self, q, k, v):
...
+            if not is_causal:
+                cu_seqlens_k = torch.arange(0, (batch_size + 1) * seqlen_k, step=seqlen_k, dtype=torch.int32,
+                                            device=q.device)
+            else:
+                cu_seqlens_k = cu_seqlens_q
 
         output = flash_attn_unpadded_func(
             q, k, v, cu_seqlens_q, cu_seqlens_k, seqlen_q, seqlen_k,
-            self.dropout_p,
+            self.dropout_p if self.training else 0.0,
```
Comment (yejunguo) [inline megatron/model/transformer.py:394]: looks that the origin code of self.dropout_p is correct.
